### DISCUSSION TITLE
Avoid camelCase/PascalCase separation

### DISCUSSION THREAD

**Discussion by samferenczi** [cantools/cantools#457] [Q&A]
Hello,

is there a possibility to NOT use camelCase/PascalCase separation during C code generation?

Thanks!

**Comment by juleq** [comment]
So you prefer utf🐍token🐍based🐍separation?

Cannot comment on the code generator as I am not a user of it ATM. Just stumbled over the apparently  missing mentioning of your preference 😜.

You could clarify your preference and then ask if a PR would have chances to be accepted.

Best regards.

**Reply by samferenczi** [reply]
Yes, you are right with the missing information :D

The best would be to leave it as it is (e.g. if the original message name is MessageName then the generated message name is also MessageName not message_name).
A worse but still a better solution for comparing/searching with the original DBC would be just lowering the characters (e.g. MessageName to messagename). I know it's more difficult to read the code but easier to search for the functions and signals.
